source/neuropod/bindings/java/src/main/java/com/uber/neuropod/LibraryLoader.java

```diff
@@ -1,19 +1,143 @@
 package com.uber.neuropod;
 
-import java.io.*;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.BufferedOutputStream;
+import java.io.FileOutputStream;
 import java.net.URL;
 import java.nio.file.Files;
-import java.nio.file.Path;
 import java.util.Arrays;
 import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
 /**
  * This class is responsible for loading jni library
  */
 class LibraryLoader {
+    private LibraryLoader() {}
+    private static final Logger LOGGER = Logger.getLogger(LibraryLoader.class.getName());
+    // C++ library names. If we use the Neruopod Java API from a jar file these files may be packed into the jar file.
+    private static final List<String> EMBEDDED_LIB_NAMES = Arrays.asList(
+                                                                // Neuropod
+                                                                "libneuropod.so",
+                                                                // Backend libaries name
+                                                                "libneuropod_tensorflow_backend.so",
+                                                                "libneuropod_torchscript_backend.so",
+                                                                "libneuropod_pythonbridge_backend.so",
+                                                                // TensorFlow libraries name
+                                                                "libtensorflow_framework.so",
+                                                                "libtensorflow.so",
+                                                                // TorchScript libraries linux name
+                                                                "libc10.so",
+                                                                "libcaffe2_detectron_ops.so",
+                                                                "libcaffe2_module_test_dynamic.so",
+                                                                "libcaffe2_observers.so",
+                                                                "libfbjni.so",
+                                                                "libgomp-7c85b1e2.so.1",
+                                                                "libpytorch_jni.so",
+                                                                "libshm.so",
+                                                                "libtorch_cpu.so",
+                                                                "libtorch_global_deps.so",
+                                                                "libtorch_python.so",
+                                                                "libtorch.so",
+                                                                // TorchScript libraries darwin name
```

Comment:
Not sure if this is critical, can it be platform specific, using compile time macro?

Reply:
We can have two list, one for linux and one for darwin.